docs/_source/tutorials_and_integrations/tutorials/feedback/end2end_examples/add-metadata-003.ipynb

```diff
@@ -0,0 +1,637 @@
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let us briefly examine what our dataset looks like. It is a dataset that consists of data items with the field `text` that is yet to be annotated."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "{'text': \"Wall St. Bears Claw Back Into the Black (Reuters) Reuters - Short-sellers, Wall Street's dwindling\\\\band of ultra-cynics, are seeing green again.\"}"
+      ]
+     },
+     "execution_count": 4,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "dataset[0].fields"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Add Metadata Properties\n",
+    "\n",
+    "Metadata properties are a great way to enhance the dataset for filtering and sorting, both in the UI and SDK. Argilla offers different types of metadata properties that you can make use of according to your needs for the specific project you are working on. For more information on metadata properties, please check out the [documentation](../../../../practical_guides/create_dataset.md#define-metadata).\n",
+    "\n",
+    "### TermsMetadataProperty\n",
+    "\n",
+    "The `TermsMetadaProperty` is a metadata property that can be used to filter the metadata of a record based on a list of possible terms or values."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "terms_metadata_property = rg.TermsMetadataProperty(name=\"group\",\n",
+    "                                                   title=\"Annotation Group\",\n",
+    "                                                   values=[\"group-1\", \"group-2\", \"group-3\"])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note that the `name` will be the name of the metadata property shown internally. `title` is again the name of the metadata property displayed in the UI. `values` will be the list of possible values for the metadata property. To add it to the dataset, we can simply use the `add_metadata_property` method."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dataset.add_metadata_property(terms_metadata_property)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let us add the metadata to individual records as well. For demonstration purposes here, we can create a list of groups in randomly select one to assign to each record. As we normally do, we should be careful to divide records into groups in equal numbers.\n",
+    "\n",
+    "We firstly learn how many records we have in the dataset and how many groups we have defined in our `TermsMetadataProperty`. Then, we create a list of group names at the number of records. "
```

Comment:
Maybe it's a phrasing thing for me, what should we understand from this?: "Then, we create a list of group names at the number of records"

Reply:
You are totally right, my bad :) The length of the list we create is the same with the length of the dataset for one-to-one mapping. I rephrased it.